inspirehep/modules/workflows/tasks/arxiv.py

```diff
@@ -68,90 +66,74 @@ def arxiv_fulltext_download(obj, eng):
     """
     arxiv_id = get_arxiv_id(obj.data)
     filename = secure_filename("{0}.pdf".format(arxiv_id))
-    if filename not in obj.files:
-        pdf = download_file_to_workflow(
-            workflow=obj,
-            name=filename,
-            url=current_app.config['ARXIV_PDF_URL'].format(
-                arxiv_id=arxiv_id
-            )
+    # We fetch if on the fly if needed
+    pdf = download_file_to_workflow(
+        workflow=obj,
+        name=filename,
+        url=current_app.config['ARXIV_PDF_URL'].format(
+            arxiv_id=arxiv_id
         )
-        pdf['doctype'] = "arXiv"
+    )
+    pdf['doctype'] = "arXiv"
+    obj.log.info("PDF retrieved from arXiv for {0}".format(arxiv_id))
 
 
 @with_debug_logging
-def arxiv_plot_extract(obj, eng):
-    """Extract plots from an arXiv archive.
+def arxiv_package_download(obj, eng):
+    """Perform the fulltext download step for arXiv records.
 
     :param obj: Workflow Object to process
     :param eng: Workflow Engine processing the object
     """
-    from wand.exceptions import DelegateError
-
     arxiv_id = get_arxiv_id(obj.data)
     filename = secure_filename("{0}.tar.gz".format(arxiv_id))
-    if filename not in obj.files:
-        tarball = download_file_to_workflow(
-            workflow=obj,
-            name=filename,
-            url=current_app.config['ARXIV_TARBALL_URL'].format(
-                arxiv_id=arxiv_id
-            )
+    tarball = download_file_to_workflow(
+        workflow=obj,
+        name=filename,
+        url=current_app.config['ARXIV_TARBALL_URL'].format(
+            arxiv_id=arxiv_id
         )
+    )
+    if tarball:
+        tarball['doctype'] = "package"
```

Comment:
Yes, this goes into the JSON: the `_files` property in the schema is there to back it, otherwise all records would become invalid due to `additionalProperties: False`.